### ISSUE TITLE
Feature Request: (metric) Time Until Next Block

### ISSUE THREAD

**Issue by yourbuddyconner** [MinaProtocol/mina#8370]
It would be great to have a Gauge metric that is the measure in seconds until the next block is to be produced. Something like `Coda_Block_producer_seconds_until_next_block`

Goal: I would like a nice way to alert when block production is imminent using Prometheus metrics.

**Comment by mrmr1993** [comment]
It would probably be easier to emit the next block production time (e.g. `Coda_Block_producer_next_block_at`), and then set the alert to trigger on `Coda_Block_producer_next_block_at - now() < threshold`. This is then just adding a hook [here](https://github.com/MinaProtocol/mina/blob/1b0be2c3b1eadb0f51814410b9e906275a6de4d4/src/lib/mina_lib/mina_lib.ml#L914)-ish and wiring in the new metric.

**Comment by yourbuddyconner** [comment]
Yeah, this is kind of hard because of the way prometheus exposes timeseries to be queried... The best example I can find of something similar (but not identical) is this countdown timer project here, which has a similar API to what I was imagining: https://github.com/tmegow/countdown-exporter

**Comment by dguess** [comment]
This would be a really useful metric for node operators. I have also had a couple of requests from others to add it to my Grafana dashboard implementation (https://grafana.com/grafana/dashboards/14299) so I think a lot of people would find it useful.

Any chance of getting this added into an upcoming release? @mrmr1993 @bkase
